test/e2e/recovery_and_restart/recovery_and_restart_test.go

```diff
@@ -133,6 +135,130 @@ var _ = ginkgo.Describe("", func() {
 		gomega.Ω(int64(resMap["vcore"].(float64))).To(gomega.Equal(core))
 	})
 
+	ginkgo.It("Verify_SleepJobs_Restart_YK", func() {
+		kClient = k8s.KubeCtl{}
+		Ω(kClient.SetClient()).To(gomega.BeNil())
+		defer restorePortForwarding(&kClient)
+
+		appID1 := normalSleepJobPrefix + "-" + common.RandSeq(5)
+		sleepPodConfig1 := k8s.SleepPodConfig{Name: "normal-sleep-job", NS: dev, Time: 20, AppID: appID1}
+		pod1, podErr := k8s.InitSleepPod(sleepPodConfig1)
+		Ω(podErr).NotTo(gomega.HaveOccurred())
+		appID2 := normalSleepJobPrefix + "-" + common.RandSeq(5)
+		sleepPodConfig2 := k8s.SleepPodConfig{Name: "normal-sleep-job-2", NS: dev, Time: 20, AppID: appID2}
+		pod2, podErr2 := k8s.InitSleepPod(sleepPodConfig2)
+		Ω(podErr2).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Submitting two normal sleep jobs")
+		job1 := k8s.InitTestJob(appID1, parallelism, parallelism, pod1)
+		_, createErr := kClient.CreateJob(job1, dev)
+		Ω(createErr).NotTo(gomega.HaveOccurred())
+		job2 := k8s.InitTestJob(appID2, parallelism, parallelism, pod2)
+		_, createErr2 := kClient.CreateJob(job2, dev)
+		Ω(createErr2).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Restart the scheduler pod immediately")
+		restartYunikorn(&kClient)
+
+		ginkgo.By("Listing pods")
+		pods, err := kClient.GetPods(dev)
+		Ω(err).NotTo(gomega.HaveOccurred())
+		fmt.Fprintf(ginkgo.GinkgoWriter, "Total number of pods in namespace %s: %d\n",
+			dev, len(pods.Items))
+		for _, pod := range pods.Items {
+			fmt.Fprintf(ginkgo.GinkgoWriter, "Pod name: %-40s\tStatus: %s\n", pod.GetName(), pod.Status.Phase)
+		}
+
+		ginkgo.By("Waiting for sleep pods to finish")
```

Comment:
Do we need to wait for these pods to finish? This adds significant delay to the test, and we could forcibly kill them off.